src/js/stores/DeploymentStore.js

```diff
@@ -25,12 +28,47 @@ function removeDeployment(deployments, deploymentId) {
   return deployments.filter(deployment => deployment.id !== deploymentId);
 }
 
+function setIsWaitingForUserDecision(deployment) {
+  deployment.currentActions.forEach(action => {
+    const results = action.readinessCheckResults;
+    action.isWaitingForUserDecision = false;
+
+    // Detect if the migration API is supported by the app and at least
+    // one migration phase is waiting for user decision
+    if (results != null && results.length > 0) {
+      action.isWaitingForUserDecision = results.some(result => {
+        if (result.lastResponse == null || result.lastResponse.body == null) {
+          return false;
+        }
+
+        try {
+          const {status} = JSON.parse(result.lastResponse.body);
+          if (status != null && status === "Waiting") {
```

Comment:
Please only test these lines in the try-/catch-block that are concerned about this.
Otherwise there could be errors suppressed accidentally.
```JavaScript
        const status;

        try {
          status = JSON.parse(result.lastResponse.body).status;
        } catch (e) { Util.noop(); }

        if (status != null && status === "Waiting") {
          const app = AppsStore.getCurrentApp(action.app);
          return !!app.hasMigrationApiSupport;
        }

        return false;
```

Same should be considered in the AppsStore portion
https://github.com/mesosphere/marathon-ui/pull/726/files#diff-183e0e82aee2602b551871df771bd11aR146